curator-framework/src/main/java/org/apache/curator/framework/imps/CuratorTempFrameworkImpl.java

```diff
@@ -98,7 +99,15 @@ private synchronized void openConnectionIfNeeded() throws Exception
         if ( cleanup == null )
         {
             ThreadFactory threadFactory = factory.getThreadFactory();
-            cleanup = (threadFactory != null) ? Executors.newScheduledThreadPool(1, threadFactory) : Executors.newScheduledThreadPool(1);
+
```

Comment:
Maybe we want a ThreadUtils.newNonDaemonThreadFactory() to replace this?
But LGTM either way.